integration/containerd/cri/integration-tests.sh

```diff
@@ -213,13 +225,58 @@ EOF
 	sleep 1
 	remained=$(ps aux|grep shimv2|grep -v grep || true)
 	[ -z $remained ] || die "found remaining shimv2 process $remained"
-	info "stop pod $podid"
-	sudo crictl stopp $podid
-	info "remove pod $podid"
-	sudo crictl rmp $podid
+
+	testContainerStop
+
 	info "stop containerd"
 }
 
+TestContainerMemoryUpdate() {
+	if [[ "${KATA_HYPERVISOR}" != "qemu" ]]; then
+		return
+	fi
+
+	test_virtio_mem=$1
+
+	if [ $test_virtio_mem ]; then
+		if [[ "$arch" != "x86_64" ]]; then
+			return
+		fi
+		info "Test container memory update with virtio-mem"
+
+		sudo sed -i -e 's/^#enable_virtio_mem.*$/enable_virtio_mem = true/g' "${kata_config}"
+	else
+		info "Test container memory update without virtio-mem"
+
+		sudo sed -i -e 's/^enable_virtio_mem.*$/#enable_virtio_mem = true/g' "${kata_config}"
+	fi
+
+	testContainerStart
+
+	vmSize=$(($(crictl exec -t -i $c cat /proc/meminfo | grep "MemTotal:" | awk '{print $2}')*1024))
+	if [ $vmSize -gt $((2*1024*1024*1024)) ] || [ $vmSize -lt $((2*1024*1024*1024-128*1024*1024)) ]; then
+		die "The VM memory size $vmSize before update is not right"
+	fi
+
+	sudo crictl update --memory $((2*1024*1024*1024)) $cid
+
+	vmSize=$(($(crictl exec -t -i $c cat /proc/meminfo | grep "MemTotal:" | awk '{print $2}')*1024))
+	if [ $vmSize -gt $((4*1024*1024*1024)) ] || [ $vmSize -lt $((4*1024*1024*1024-128*1024*1024)) ]; then
+		die "The VM memory size $vmSize after increase is not right"
+	fi
+
+	if [ $test_virtio_mem ]; then
+		sudo crictl update --memory $((1*1024*1024*1024)) $cid
+
+		vmSize=$(($(crictl exec -t -i $c cat /proc/meminfo | grep "MemTotal:" | awk '{print $2}')*1024))
+		if [ $vmSize -gt $((3*1024*1024*1024)) ] || [ $vmSize -lt $((3*1024*1024*1024-128*1024*1024)) ]; then
+			die "The VM memory size $vmSize after decrease is not right"
```

Comment:
Should ensure the pod to be deleted(by calling `testContainerStop`) even if the assertion is failed.

Reply:
Fixed.